From: Jan Schmidt <list.btrfs@jan-o-sch.net>
Date: Wed, 4 Jul 2012 08:15:02 -0600
Patch-mainline: 3.6
Git-commit: e6466e354a5c23717325adecf387f93be4b9c830
Git-commit: cf5388307a2b4faab4b11d732b61c85741be6169
References: FATE#312888
Subject: [PATCH] Btrfs: fix buffer leak in btrfs_next_old_leaf

When calling btrfs_next_old_leaf, we were leaking an extent buffer in the
rare case of using the deadlock avoidance code needed for the tree mod log.

Signed-off-by: Jan Schmidt <list.btrfs@jan-o-sch.net>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/ctree.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/btrfs/ctree.c b/fs/btrfs/ctree.c
index 8206b39..67fe46f 100644
--- a/fs/btrfs/ctree.c
+++ b/fs/btrfs/ctree.c
@@ -5127,6 +5127,7 @@ again:
 				 * locked. To solve this situation, we give up
 				 * on our lock and cycle.
 				 */
+				free_extent_buffer(next);
 				btrfs_release_path(path);
 				cond_resched();
 				goto again;
-- 
1.8.0

